Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -1269,6 +1269,53 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     return handleEffectSpecifiers(node)
   }
 
+  public override func visit(_ node: TypeInheritanceClauseSyntax) -> SyntaxVisitorContinueKind {
+    if shouldSkip(node) {
+      return .skipChildren
+    }
+
+    if let unexpected = node.unexpectedBeforeColon,
+      let leftParen = unexpected.tokens(satisfying: { $0.tokenKind == .leftParen }).first
```

Comment:
Since you are making all the unexpected nodes missing, I think this should be 

```suggestion
      let leftParen = unexpected.onlyToken(where: { $0.tokenKind == .leftParen })
```